packages/malloy/src/lang/grammar/MalloyParser.g4

```diff
@@ -32,28 +32,38 @@ malloyStatement
   | defineQuery
   | importStatement
   | runStatement
+  | docAnnotations
+  | ignoredObjectAnnotations
   ;
 
 defineSourceStatement
-  : SOURCE sourcePropertyList
+  : tags SOURCE sourcePropertyList
   ;
 
 defineQuery
-  : QUERY topLevelQueryDefs      # namedQueries_stub
-  | QUERY topLevelAnonQueryDef   # anonymousQuery
+  : topLevelQueryDefs                 # use_top_level_query_defs
+  | tags QUERY topLevelAnonQueryDef   # anonymousQuery
   ;
 
 topLevelAnonQueryDef
-  : query
+  : tags query
+  ;
+
+tags
+  : ANNOTATION*
+  ;
+
+isDefine
+  : beforeIs=tags IS afterIs=tags
   ;
 
 runStatement
-  : RUN topLevelAnonQueryDef     # runStatementDef
-  | RUN queryName                # runStatementRef
+  : blockTags=tags RUN noteTags=tags topLevelAnonQueryDef     # runStatementDef
+  | RUN queryName                                             # runStatementRef
```

Comment:
No tags for `run: queryName` because that rule goes away when `queryName` is a legal `topLevelAnonQueryDef` ... it it also 10 lines of code if this bothers you